web/pages/api/og-image/[slug].ts

```diff
@@ -0,0 +1,42 @@
+import Jimp from 'jimp-compact';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { resolve } from 'path';
+import wrap from 'word-wrap';
+
+// Make Vercel include this file (referenced by .fnt below) in lambda
+resolve('public', 'static', 'fonts', 'national-2-narrow-bold.png');
+
+const FONT_SIZE = 80;
+const PADDING_X = 32;
+const PADDING_Y = 40;
+const WORD_WRAP_LIMIT = 26;
+
+export default async function ogImage(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  try {
+    // Resolves to root of /web directory, __dirname is buggy in Next.js
+    const imagePath = resolve('images', 'og_image_background.png');
+    const image = await Jimp.read(imagePath);
+
+    const font = await Jimp.loadFont(
+      resolve('public', 'static', 'fonts', 'national-2-narrow-bold.fnt')
+    );
+    wrap(req.query.slug as string, { width: WORD_WRAP_LIMIT })
+      .split('\n')
+      .forEach((line, i) =>
+        image.print(font, PADDING_X, PADDING_Y + i * FONT_SIZE, line)
+      );
```

Comment:
I don't understand why this is wrapped based on number of characters. This means the lines can get extra "jagged" if we have bad "luck" with the string contents, since the font is not fixed width. Also it doesn't prevent the text from bleeding over the shapes on the right, which the Figma sketches do try to avoid.

Is it because of the line height? And if so, could it be an option to modify the .fnt file's `lineHeight` instead so that Jimp's own text wrapping can be used instead of the `wrap` + `split` + `forEach` combination?

E.g. `image.print(font, PADDING_X, PADDING_Y, req.query.slug as string, 752);` (where 752 can be tweaked if necessary, and of course be defined in a constant)

Reply:
I didn't study `jimp`'s docs closely enough to notice that it had a built-in word-wrapping option. Good catch! 07edc7d.

Reply:
OK.

As mentioned, it seems to be possible to adjust `lineHeight=96` to e.g. `lineHeight=77` in the .fnt file to decrease the line height when there are many lines, e.g. https://structured-content-2022-web-git-generate-og-image.sanity.build/api/og-image/How%20Publishing%20Organizations%20Harness%20Structured%20Content%20to%20Power%20Editorial%20Experiences.

But this is just based on experiment, I haven't read up on the specs for .fnt if there is such a thing.

Reply:
Good suggestion, that looks much better. 2584d37